Python/_warnings.c

```diff
@@ -440,6 +479,7 @@ warn_explicit(PyObject *category, PyObject *message,
     PyObject *key = NULL, *text = NULL, *result = NULL, *lineno_obj = NULL;
     PyObject *item = NULL;
     PyObject *action;
+    PyFrameObject *f = PyThreadState_GET()->frame;
```

Comment:
please move it into the print_frames() block, it's only needed there. Or even move it into print_frames() function.

You should use _PyThreadState_GET() to ensure that you get the optimized macro ;-)

Reply:
My C is rusty, so apologies for potentially simple questions: 
isn't that the same because pycore_pystate aliases it ? 

```
Include/internal/pycore_pystate.h:#define PyThreadState_GET() _PyThreadState_GET()
```

Also Im' confused about the uppercase vs lowercase `PyThreadState(_Get/_GET)`.

Reply:
You only get the alias if you include pycore_pystate. I prefer to always use _PyThreadState_GET() to ensure to always get the optimized macro.